src/main/java/com/jakubroks/quiz/service/AuthService.java

```diff
@@ -0,0 +1,28 @@
+package com.jakubroks.quiz.service;
+
+import com.jakubroks.quiz.entity.User;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class AuthService {
+
+    private final UserService userService;
+
+    public AuthService(UserService userService) {
+        this.userService = userService;
+    }
+
+    public ResponseEntity<Object> requireLogin(String key, Function<User, ResponseEntity<Object>> ok) {
```

Comment:
jak jak na zajęciach zaimplementuj opcję z errorem (jak dobrze pamiętam) - to co robiliśmy na zajęciach aby unikać lambdy i popraw wszędzie